3.8/arangosearch-analyzers.md

```diff
@@ -489,8 +489,16 @@ be run on DB-Servers in case of a cluster deployment. User-defined functions
 are not permitted.
 
 The input data is provided to the query via a bind parameter `@param`.
-It is always a string. The query result should be a string, `null` or
-an array of strings (and `null`s).
+It is always a string. The query result will be converted according to
+`returnType` parameter value. 
+
+{% hint 'information' %}
+If `returntType` is `"number"` or `"bool"` there is no need 
+to use custom analyzer via `ANALYZER()` function in your queries. Numerics and booleans
+will be available the same way as if these values were directly in the indexed documents.
```

Comment:
Does this try to explain the following comment from the feature PR?

> automatically apply internal numeric/boolean analyzers for tokenizing corresponding values

It's the first time I hear about internal number/bool analyzers. Do we really need to explain this if it's hidden from the user and just works as expected?

Or does this refer to something completely different? That users should not create AQL Analyzers only to index plain numbers or booleans, because that works out of the box with any Analyzer?

Reply:
Well, by internal conversion rules I wanted to tell about corresponding type conversion functions e.g. TO_STRING

Reply:
Okay, so if I use an AQL Analyzer with the "number" returnType for instance, then I don't have to set a context Analyzer in the query to match that AQL Analyzer. I can simply compare the value directly, like `SEARCH doc.value == 42` (defaults to identity, but Analyzer doesn't matter because processing is skipped for booleans and numbers). Using `SEARCH ANALYZER(doc.value == 42, "my_aql_analyzer_with_number_return_type")` would work as well, but is unnecessary - is that correct?

Reply:
Yes